idseq_dag/engine/pipeline_flow.py

```diff
@@ -267,3 +272,25 @@ def start(self):
                 log.write("An exception was thrown. Stage failed.")
                 raise e
         log.write("all steps are done")
+
+
+def _get_name_from_path(dag_json):
+    """
+    Returns a useful stage name from a dag_json file page for when the dag_json
+    is missing an explicit name.
+
+    >>> _get_name_from_path('templates/gsnap_index.json')
+    'gsnap_index'
+
+    >>> _get_name_from_path('gsnap_index.json')
+    'gsnap_index'
+
+    >>> _get_name_from_path('templates/gsnap_index')
+    'gsnap_index'
+    """
+    return dag_json.split('/')[-1].split('.')[0]
```

Comment:
`return os.path.splitext(os.path.basename(dag_json))[0]` to be more robust?

Reply:
good idea